src/asciidoctor-extensions-api.js

```diff
@@ -257,6 +256,208 @@ Registry.prototype.docinfoProcessor = function (name, processor) {
+ */
+Registry.prototype.getIncludeProcessors = function () {
+  return this['$include_processors']()
+}
+
+/**
+ * Retrieves the Extension proxy objects for all the {{@link Extensions/Postprocessor}} instances stored in this registry.
+ *
+ * @memberof Extensions/Registry
+ * @returns an {array} of Extension proxy objects.
+ */
+Registry.prototype.getPostprocessors = function () {
+  return this['$postprocessors']()
+}
+
+/**
+ * Retrieves the Extension proxy objects for all the {{@link Extensions/DocinfoProcessor}} instances stored in this registry.
+ *
+ * @memberof Extensions/Registry
+ * @param location - A {string} for selecting docinfo extensions at a given location (head or footer) (default: undefined)
+ * @returns an {array} of Extension proxy objects.
+ */
+Registry.prototype.getDocinfoProcessors = function (location) {
+  return this['$docinfo_processors'](location)
+}
+
+/**
+ * Retrieves the Extension proxy objects for all the {{@link Extensions/BlockProcessor}} instances stored in this registry.
+ *
+ * @memberof Extensions/Registry
+ * @returns an {array} of Extension proxy objects.
+ */
+Registry.prototype.getBlocks = function () {
+  return this.block_extensions['$values']()
+}
+
+/**
+ * Retrieves the Extension proxy objects for all the {{@link Extensions/BlockMacroProcessor}} instances stored in this registry.
+ *
+ * @memberof Extensions/Registry
+ * @returns an {array} of Extension proxy objects.
+ */
+Registry.prototype.getBlockMacros = function () {
+  return this.block_macro_extensions['$values']()
+}
+
+/**
+ * Retrieves the Extension proxy objects for all the {{@link Extensions/InlineMacroProcessor}} instances stored in this registry.
+ *
+ * @memberof Extensions/Registry
+ * @returns an {array} of Extension proxy objects.
+ */
+Registry.prototype.getInlineMacros = function () {
+  return this['$inline_macros']()
+}
+
+/**
+ * Get any {{@link Extensions/InlineMacroProcessor}} extensions are registered to handle the specified inline macro name.
+ *
+ * @param name - the {string} inline macro name
+ * @memberof Extensions/Registry
+ * @returns the Extension proxy object for the {{@link Extensions/InlineMacroProcessor}} that matches the inline macro name or undefined if no match is found.
+ */
+Registry.prototype.getInlineMacroFor = function (name) {
+  var result = this['$registered_for_inline_macro?'](name)
+  return result === false ? undefined : result
+}
+
+/**
+ * Get any {{@link Extensions/BlockProcessor}} extensions are registered to handle the specified block name appearing on the specified context.
+ * @param name - the {string} block name
+ * @param context - the context of the block: paragraph, open... (optional)
+ * @memberof Extensions/Registry
+ * @returns the Extension proxy object for the {{@link Extensions/BlockProcessor}} that matches the block name and context or undefined if no match is found.
+ */
+Registry.prototype.getBlockFor = function (name, context) {
```

Comment:
The function returns an extension or undefined if not found. 

In Asciidoctor core, the method `find_block_extension` returns the extension or nil if not found and `registered_for_block?` returns the extension or false if not found.

I believe it's pretty uncommon in JavaScript to return a boolean or the found object. I think it's better to return undefined/null or the found object. If we want to return a boolean value we could also introduce a `hasBlockFor` that returns true if found and otherwise false. (in my opinion it's unnecessary)